Modules/_cursesmodule.c

```diff
@@ -3968,48 +4009,28 @@ _curses_qiflush_impl(PyObject *module, int flag)
  * and _curses.COLS */
 #if defined(HAVE_CURSES_RESIZETERM) || defined(HAVE_CURSES_RESIZE_TERM)
 static int
-update_lines_cols(void)
+update_lines_cols(PyObject *private_module)
 {
-    PyObject *o;
-    PyObject *m = PyImport_ImportModule("curses");
-
-    if (!m)
-        return 0;
-
+    PyObject *exposed_module = NULL, *o = NULL;
+    exposed_module = PyImport_ImportModule("curses");
+    CHECK_NOT_NULL_OR_ERROR(exposed_module);
     o = PyLong_FromLong(LINES);
-    if (!o) {
-        Py_DECREF(m);
-        return 0;
-    }
-    if (PyObject_SetAttrString(m, "LINES", o)) {
-        Py_DECREF(m);
-        Py_DECREF(o);
-        return 0;
-    }
-    if (PyDict_SetItemString(ModDict, "LINES", o)) {
-        Py_DECREF(m);
-        Py_DECREF(o);
-        return 0;
-    }
+    CHECK_NOT_NULL_OR_ERROR(o);
+    CHECK_RET_CODE_OR_ERROR(PyObject_SetAttrString(exposed_module, "LINES", o));
+    CHECK_RET_CODE_OR_ERROR(PyObject_SetAttrString(private_module, "LINES", o));
```

Comment:
I would expect a macro getting a module, a key and a value. Rather than a macro taking a function call.